packages/ng-query/src/lib/tests/query.spec.ts

```diff
@@ -0,0 +1,138 @@
+      staleTime: 1000,
+      queryKey: ['key0'],
+      queryFn: expect.any(Function), // TODO: make it work with simpleFetcher function
+    });
+  });
+
+  it('should return loading status initially', () => {
+    const query = useQuery(['key1'], simpleFetcher);
+
+    expect(query.getCurrentResult()).toMatchObject({
+      status: 'loading',
+      isLoading: true,
+      isFetching: true,
+      isStale: true,
+    });
+  });
+
+  it('should resolve to success: useQuery(key, dataFn)', async () => {
+    const query = useQuery(['key2'], fetcher('result2'));
+
+    const observerSpy = subscribeSpyTo(query.result$);
+    await flushPromises();
+    const [loading, success] = observerSpy.getValues();
+    expect(loading.status).toBe('loading');
+    expect(success).toMatchObject({
+      status: 'success',
+      data: 'result2',
+      isLoading: false,
+      isFetching: false,
+      isFetched: true,
+      isSuccess: true,
+    });
+  });
+
+  it('should resolve to success: useQuery(optionsObj)', async () => {
+    const query = useQuery({
+      queryKey: ['key31'],
+      queryFn: fetcher('result31'),
+      enabled: true,
+    });
+
+    const observerSpy = subscribeSpyTo(query.result$);
+    await flushPromises();
+    const [loading, success] = observerSpy.getValues();
+    expect(loading.status).toBe('loading');
+    expect(success).toMatchObject({
+      status: 'success',
+      data: 'result31',
+      isLoading: false,
+      isFetching: false,
+      isFetched: true,
+      isSuccess: true,
+    });
+  });
+
+  it('should resolve to success: useQuery(key,optionsObj)', async () => {
+    const query = useQuery(['key32'], {
+      queryFn: fetcher('result32'),
+      enabled: true,
+    });
+
+    const observerSpy = subscribeSpyTo(query.result$);
+    await flushPromises();
+    const [loading, success] = observerSpy.getValues();
+    expect(loading.status).toBe('loading');
+    expect(success).toMatchObject({
+      status: 'success',
+      data: 'result32',
+      isLoading: false,
+      isFetching: false,
+      isFetched: true,
+      isSuccess: true,
+    });
+  });
+
+  it.skip('should reject if queryFn errors out', async () => {
+    const query = useQuery(['key3'], () => {
+      return throwError('error here');
+    });
+
+    const observerSpy = subscribeSpyTo(query.result$);
+
+    await flushPromises();
+    const [loading, error] = observerSpy.getValues();
+    expect(loading.status).toBe('loading');
+
+    expect(error).toMatchObject({
+      status: 'error',
+      data: undefined,
+      error: { message: 'Some error' },
+      isLoading: false,
+      isFetching: false,
+      isFetched: true,
+      isError: true,
+      failureCount: 1,
+      failureReason: { message: 'Some error' },
+    });
+  });
+
+  it.skip('should call onSuccess callback', async () => {
+    const onSuccess = jest.fn();
+
+    useQuery(['key6'], simpleFetcher, {
+      onSuccess: () => {
```

Comment:
You're not using the spy. It should be:

```ts
onSuccess,
```

Reply:
here I was testing if it gets logged, but it does not. I am having no idea why its not being called in tests but it does in playground.